### PR TITLE
*: add more stats related diagnostic information to execution plan and slow log | tidb-test=pr/2123

### LINKED ISSUE #43122: Add more stats related diagnostic information to execution plan and slow log
### Current stats-related diagnostic information
For now, only the table level stats version is collected.

#### operator info field of EXPLAIN output
If the stats on this table is unavailable, or the stats is outdated (this behavior is disabled by default now), there will be “stats: pseudo” on the table/index scan operator.

#### Stats field in the slow log
Record all stats versions of the table related to all table/index scan. The output format is “table name: [pseudo | stats version]”.
For example:
\# Stats: t1:pseudo,t2:434264434430967808

### Extend the output information
Based on the current output information, extend the output to record the column/index level loading status of the stats.

#### Record the status of used column/index stats
In `GetRowCountByXXX` (which is mainly the stats used), for non-pseudo stats, record the used column/index stats and its loading status.

#### operator info field of EXPLAIN output
Besides current information, if the recorded information contains stats that are not fully loaded or are evicted, it will output “stats: partial”, followed by their corresponding status in the format like “[col1: notInitialized, col2: allEvicted]”.
It would output information for 3 columns/indexes at most, the rest will become “...” plus an aggregated information to show how many stats are in which status.
For example:
stats: partial [col1: unInitialized]
stats: partial [col1: unInitialized, col2: onlyCmsEvicted, idx4: onlyCmsEvicted]
stats: partial [col1: unInitialized, col5: onlyCmsEvicted, idx2: allEvicted…(more: 2 onlyCmsEvicted, 3 allEvicted)]

#### Stats field in the slow log
After the current “pseudo | stats version”, record the table level row count and modify count, the print the status of stats that are not fully loaded or are evicted in the format like “[col1: unInitialized, col2: AllEvicted]”.
For example:
\# Stats: t1:pseudo[10000;0], t2:434264434430967808[500000;0][col1: unInitialized], t3 p2:434264434430967808[101000;1000][col1: unInitialized, col2: onlyCmsEvicted][idx4: onlyCmsEvicted], t4 global:434264434430967808[150000;2000][col1: unInitialized]

### PR DESCRIPTION
**Pull request by time-and-fate**
### What problem does this PR solve?

Issue Number: close #43122

### What is changed and how it works?

- Remove the previous fields and methods that were used for the `Stats` field of the slow log
  - `StatementContext.StatsLoadStatus`
  - `SlowQueryLogItems.StatsInfos`
  - `SlowQueryLogItems.StatsLoadStatus`
  - `writeStatsLoadStatusItems()`
  - `convertStatusIntoString()`
- Add several new fields and methods to maintain the information for the `Stats` field of the slow log and the `stats: ` information in the execution plan
  - `UsedStatsInfoForTable`
    - `FormatForExplain()`
    - `WriteToSlowLog()`
  - `SlowQueryLogItems.UsedStats`
  - `StatementContext.usedStatsInfo`
  - `PhysicalIndexScan.usedStatsInfo`
  - `PhysicalTableScan.usedStatsInfo`
- The maintained information is more than before
  - The partition name is added after the table name ("global" or partition name)
  - The realtime row count and modify count are added
  - The "missing" state is added for the case where the column/index stats is not in `HistColl`
- The output changes can be found in the issue

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
image
image
image
image
image
image

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: sessionctx/stmtctx/stmtctx.go [hunk 1]
```diff
@@ -1259,6 +1264,142 @@ func (d *CopTasksDetails) ToZapFields() (fields []zap.Field) {
...
+	ColumnStatsLoadStatus map[int64]string
+	IndexStatsLoadStatus  map[int64]string
+}
+
+// FormatForExplain format the content in the format expected to be printed in the execution plan.
+// case 1: if stats version is 0, print stats:pseudo.
+// case 2: if stats version is not 0, and there are column/index stats that are not full loaded,
+// print stats:partial, then print status of 3 column/index status at most. For the rest, only
+// the count will be printed, in the format like (more: 1 onlyCmsEvicted, 2 onlyHistRemained).
+func (s *UsedStatsInfoForTable) FormatForExplain() string {
```
Comment (qw4990) [inline sessionctx/stmtctx/stmtctx.go:1304]: Better to add some test cases(examples) for this function.

**Review by xuyifangreeneyes — APPROVED** [review]

**Pull request merged**